Mathlib/CategoryTheory/Localization/DerivabilityStructure/Derives.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Localization.DerivabilityStructure.PointwiseRightDerived
+
+/-!
+# Deriving functors using a derivability structure
+
+Let `Φ : LocalizerMorphism W₁ W₂` be a localizer morphism between classes
+of morphisms on categories `C₁` and `C₂`. Let `F : C₂ ⥤ H`.
+When `Φ` is a left or right derivability structure, it allows to derive
+the functor `F` (with respect to `W₂`) when `Φ.functor ⋙ F : C₁ ⥤ H`
+inverts `W₁` (this is the most favorable case when we can apply the lemma
+`hasPointwiseRightDerivedFunctor_iff_of_isRightDerivabilityStructure`).
+We define `Φ.Derives F` as an abbreviation for `W₁.IsInvertedBy (Φ.functor ⋙ F)`.
+
+When `h : Φ.Derives F` holds and `Φ` is a right derivability structure,
+we show that `F` has a right derived functor with respect to `W₂`.
+Under this assumption, if `L₂ : C₂ ⥤ D₂` is a localization functor
+for `W₂`, then a functor `RF : D₂ ⥤ H` equipped with a natural
+transformation `α : F ⟶ L₂ ⋙ RF` is the right derived functor of `F` iff
+for any `X₁ : C₁`, the map `α.app (Φ.functor.obj X₁)` is an isomorphism.
+
+-/
+
+universe v₁ v₂ v₃ v₄ u₁ u₂ u₃ u₄
+
+namespace CategoryTheory
+
+open Limits Category
+
+variable {C₁ : Type u₁} {C₂ : Type u₂} {H : Type u₃}
+  [Category.{v₁} C₁] [Category.{v₂} C₂] [Category.{v₃} H]
+  {D₂ : Type u₄} [Category.{v₄} D₂]
+  {W₁ : MorphismProperty C₁} {W₂ : MorphismProperty C₂}
+
+namespace LocalizerMorphism
+
+variable (Φ : LocalizerMorphism W₁ W₂) (F : C₂ ⥤ H)
+
+/-- Given a localizer morphism `Φ : LocalizerMorphism W₁ W₂` between
+morphism properties on `C₁` and `C₂`, and a functor `C₂ ⥤ H`, this
+is the property that `W₁` is inverted by `Φ.functor ⋙ F`.
+In case `Φ` is a (left/right) derivability structure, this allows
+the construction of a derived functor for `F` relatively to `W₂`. -/
+abbrev Derives : Prop := W₁.IsInvertedBy (Φ.functor ⋙ F)
+
+namespace Derives
+
+variable (h : Φ.Derives F) [Φ.IsRightDerivabilityStructure]
+
+include h
+
+lemma hasPointwiseRightDerivedFunctor : F.HasPointwiseRightDerivedFunctor W₂ := by
+  rw [hasPointwiseRightDerivedFunctor_iff_of_isRightDerivabilityStructure Φ F]
+  exact Functor.hasPointwiseRightDerivedFunctor_of_inverts _ h
+
+section
+
+variable {L₂ : C₂ ⥤ D₂} [L₂.IsLocalization W₂] {RF : D₂ ⥤ H} (α : F ⟶ L₂ ⋙ RF)
+
+lemma isIso_α (X₁ : C₁) [RF.IsRightDerivedFunctor α W₂] :
+    IsIso (α.app (Φ.functor.obj X₁)) := by
```

Comment:
I'm not sure if `α` should be in the name as it is a variable name and not some definition appearing in the statement.